File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        output = []

        def inorder(rootNode):
            if not rootNode:
                return

            inorder(rootNode.left)
            output.append(rootNode.val)
            inorder(rootNode.right)

        inorder(root)
        return output

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
    result := []int{}
    var dfs func(node *TreeNode)
    dfs = func (root *TreeNode) {
        if root == nil {
            return
        }
        dfs(root.Left)
        result = append(result, root.Val)
        dfs(root.Right)
    }
    dfs(root)
    return result
}